Remove usages of shell=True in AWS provisioner (resolves #1183)

cket opened the pull request.

cket commented:
jenkins, test this please

cket commented:
@hannes-ucsc ready for review

cket commented:
Jenkins, test this please

cket commented:
Jenkins, test this please

cket commented:
@hannes-ucsc ready for review - let me know if have ideas for more special character tests! I think I covered the basics we talked about.

cket commented:
jenkins, test this please

cket commented:
jenkins, test this please

cket commented:
jenkins, test this please

cket commented:
@hannes-ucsc ready for review

hannes-ucsc commented on src/toil/test/utils/utilsTest.py:
```diff
@@ -79,14 +80,27 @@ def testAWSProvisionerUtils(self):
+            compareTo = "import sys; assert sys.argv[1]=='$PATH'"
+            AWSProvisioner.sshLeader(clusterName=clusterName,
+                                     args=['python', '-', '$PATH'],
+                                     input=compareTo)
```
This could be a loop that computes `compareTo` dynamically using `%r`.

I would also test

```
args=['python', '-', '"testString"']
args=['python', '-', '"'] # one double quote character
args=['python', '-', "`"] # one single quote character
args=['python', '-', '\\'] # one single single backslash character
```

hannes-ucsc replied:
I just edited the above comment.

hannes-ucsc commented on src/toil/test/utils/utilsTest.py:
```diff
@@ -89,18 +91,25 @@ def testAWSProvisionerUtils(self):
+                           '; cat'
+                           ]
+            for test in testStrings:
+                logger.info('Testing SSH with special string: %s', test)
```
`%r` not `%s`

cket commented:
jenkins, test this please

cket commented:
jenkins, test this please

cket commented:
Looks like a pip outage... 
jenkins, test this please

cket commented:
I've addressed all of @hannes-ucsc comments and the code I've added since his last review is simple and has a test so I'm sure it works. Since Github complains when I try to approve my own PR I'll hold off on merging this for now, but it may start blocking #1300. At that point I'll go ahead and merge the PR anyway.
